### PR TITLE
add back the silent kwarg to message.delete

### PR DESCRIPTION
**Pull request by TheMoksej**
## Summary

Just adding back the old feature to 2.0 :mmlool:

## Checklist

- [ ] If code changes were made then they have been tested.
    - [ ] I have updated the documentation to reflect the changes.
- [ ] This PR fixes an issue.
- [ ] This PR adds something new (e.g. new method or parameters).
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [ ] This PR is **not** a code change (e.g. documentation, README, ...)

### CODE REVIEW

**Comment by justanotherbyte** [comment]
```py
            except Exception as e:
                if silent:
                    pass
                else:
                    raise e
```
can be shortened to:
```py
            except Exception as e:
                if not silent:
                    raise e
```

**Comment by TheMoksej** [comment]
> ```python
>             except Exception as e:
>                 if silent:
>                     pass
>                 else:
>                     raise e
> ```
> 
> can be shortened to:
> 
> ```python
>             except Exception as e:
>                 if not silent:
>                     raise e
> ```

That's already done [2bc75d5](https://github.com/iDevision/enhanced-discord.py/pull/9/commits/2bc75d51eadbd10fa357fa064d88a14f67393afe)

**Comment by justanotherbyte** [comment]
ah, my bad

**Comment by TheMoksej** [comment]
Nah, it's my bad for not noticing it in the first place, just kinda copy-pasted the old code.

File: discord/message.py [hunk 1]
```diff
@@ -1144,7 +1149,11 @@ async def delete(delay: float):
 
             asyncio.create_task(delete(delay))
         else:
-            await self._state.http.delete_message(self.channel.id, self.id)
+            try:
+                await self._state.http.delete_message(self.channel.id, self.id)
+            except Exception as e:
+                if not silent:
+                    raise e
```
Comment (NiumXp) [inline discord/message.py:1156]: You are not using the error, so, you can do
```py
            except Exception:
                if not silent:
                    raise
```

Comment (TheMoksej) [reply discord/message.py:1156]: Good point, let me change that

File: discord/message.py [hunk 2]
```diff
@@ -1106,7 +1106,7 @@ def system_content(self):
         if self.type is MessageType.guild_invite_reminder:
             return 'Wondering who to invite?\nStart by inviting anyone who can help you build the server!'
 
-    async def delete(self, *, delay: Optional[float] = None) -> None:
+    async def delete(self, *, delay: Optional[float] = None, silent: Optional[bool] = False) -> None:
```
Comment (NiumXp) [inline discord/message.py:1109]: The use of `Optional` is wrong.
```py
    async def delete(self, *, delay: Optional[float] = None, silent: bool = False) -> None:
```
If a parameter has a default value different to `None` (and use it for anything), it is not an optional parameter.

**Review by IAmTomahawkx — APPROVED** [review]

**Pull request merged**